### ISSUE TITLE
Exception when using DoctrineOrmMappingsPass::createAttributeMappingDriver together with doctrine/orm > v3

### ISSUE THREAD

**Issue by uerka** [doctrine/DoctrineBundle#1844]
### Bug Report

Not sure if i should put it here or into doctrine/orm as it seems like compatibility problem.

|    Q                                              |   A
|-------------------------------| ------
| doctrine\doctrine-bundle         | 2.13.1
| doctrine\orm                              | 3.3.0

#### Summary

We have a private bundle, that we're reusing on multiple projects. This bundle provides some entities with mapping. Within this bundle we're calling DoctrineOrmMappingsPass to register bundle's entities mapping:  

```php 
$container->addCompilerPass(DoctrineOrmMappingsPass::createAttributeMappingDriver($namespaces, $directories));
```

#### Current behavior

Currently app crashes hard with  \InvalidArgumentException 

`The $reportFieldsWhereDeclared argument is no longer supported, make sure to omit it when calling Doctrine\ORM\Mapping\Driver\AttributeDriver::__construct.`

#### Expected behavior

It probably should not crash)
As soon as its going to be omitted by doctrine\orm we would like to avoid pass reportFieldsWhereDeclared from bundle.

#### How to reproduce

Its quite simple: 
1)  doctrine/orm version > 3
2) latest doctrine/doctrine-bundle
3) call DoctrineOrmMappingsPass to add mapping by attributes - for example like following:

```php 
namespace My\CustomBundle;

use Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\DoctrineOrmMappingsPass;
use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\HttpKernel\Bundle\Bundle;

class MyCustomBundle extends Bundle
{
    public function build(ContainerBuilder $container): void
    {
        $namespaces = [
            'My\CustomBundle\Model',
        ];

        $directories = [
            __DIR__ . '/src/Model',
        ];
        
        $container->addCompilerPass(DoctrineOrmMappingsPass::createAttributeMappingDriver($namespaces, $directories));
     }
}
```

**Comment by greg0ire** [comment]
I believe the offending piece of code is here: https://github.com/doctrine/DoctrineBundle/blob/75026f2e42f7ab93a4d4cfb766cb69b50dd99240/src/DependencyInjection/Compiler/DoctrineOrmMappingsPass.php#L158

It should be possible to detect ORM 3 and omit that argument based when relevant.

**Comment by bobvandevijver** [comment]
Ah, just hit this as well when upgrading a dependency as well. I've submitted a PR to fix this, see #2120.

**Comment by greg0ire** [comment]
@bobvandevijver I don't think it can be an issue with the annotation driver, since it no longer exists with ORM 3. Calling `createAnnotationMappingDriver` with ORM 3 probably results in a crash already.

**Comment by bobvandevijver** [comment]
Ah, that is actually a great point, I missed that as you linked to the annotation line before. I will adjust the PR to only include the fix for the attribute driver :+1:
